### PR TITLE
Default initialize controller vendor ID

### PR DESCRIPTION
**Pull request by mrjerryjohns**
#### Problem

The Python REPL was failing to commission a device when built with detail logging disabled.

### Cause

It was not correctly initializing the `Controller::SetupParams::controllerVendorId` when creating a commissioner instance. This resulted in it using garbage values that sometimes, were correct and not zero.

### Fix

As per discussion here, this default initializes that field to `VendorId::Unspecified` as well as adding an early check for a valid value at controller creation time. This ensures that applications creating commissioners are forced to explicitly provide a value for their vendor ID. It also ensures we catch applications who don't do so early at commissioner instantiation time.

### Testing

Validate that the REPL and REPL tests pass correctly.
For other platforms, will have to check the various PR tests and see which ones fail.

### CODE REVIEW

File: src/controller/CHIPDeviceControllerFactory.cpp [hunk 1]
```diff
@@ -303,9 +303,12 @@ CHIP_ERROR DeviceControllerFactory::SetupController(SetupParams params, DeviceCo
 CHIP_ERROR DeviceControllerFactory::SetupCommissioner(SetupParams params, DeviceCommissioner & commissioner)
 {
     VerifyOrReturnError(mSystemState != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(params.controllerVendorId != VendorId::Unspecified, CHIP_ERROR_INVALID_ARGUMENT);
```
Comment (bzbarsky-apple) [inline src/controller/CHIPDeviceControllerFactory.cpp:306]: Should put this check in `DeviceControllerFactory::SetupController` too, right?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/CHIPDeviceControllerFactory.cpp (hunk 1 above)
Comment (mrjerryjohns) [reply src/controller/CHIPDeviceControllerFactory.cpp:306]: I didn't put it there since controllerVendorId only seems to be necessary for commissioners?

Comment (mrjerryjohns) [reply src/controller/CHIPDeviceControllerFactory.cpp:306]: Was trying to minimize the required set of bits controllers vs. commissioners needed to specify to be functional.

**Review by yunhanw-google — APPROVED** [review]

**Review by chrisdecenzo — APPROVED** [review]

File: src/controller/CHIPDeviceControllerFactory.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/controller/CHIPDeviceControllerFactory.cpp:306]: It's used in two places:

1. To set `mVendorId`, which seems to only be used in `DeviceCommissioner::SendOperationalCertificate`.
2. To set up the fabric table entry, which then might get used by anyone.

**Review by tehampson — APPROVED** [review]

**Review by agners — APPROVED** [review]

File: src/controller/CHIPDeviceControllerFactory.cpp (hunk 1 above)
Comment (mrjerryjohns) [reply src/controller/CHIPDeviceControllerFactory.cpp:306]: Fair enough. Will update `SetupController` too.

**Pull request merged**